connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/adhoctoolsschema/feel/FeelInputParamExtractor.java

```diff
@@ -118,14 +119,19 @@ private FeelInputParam fromFunctionInvocationParams(
   private String parameterName(Exp value) {
     if (!(value instanceof Ref valueRef)) {
       throw new FeelInputParamExtractionException(
-          "Expected parameter 'value' to be a reference, but got '%s'"
-              .formatted(value != null ? value.getClass().getSimpleName() : null));
+          "Expected parameter 'value' to be a reference (e.g. 'toolCall.customParameter'), but received %s."
+              .formatted(
+                  switch (value) {
+                    case ConstString stringResult -> "string '%s'".formatted(stringResult.value());
+                    default -> value;
+                  }));
     }
 
-    if (valueRef.names() == null || valueRef.names().size() < 2) {
+    if (valueRef.names() == null || valueRef.names().isEmpty()) {
```

Comment:
This was too restrictive and very much focuses on how our current examples are structured. Now we just expect a reference.